### ISSUE TITLE
L027 error finding table aliases when mixing how table aliases are used in joins

### ISSUE THREAD

**Issue by kochalex** [sqlfluff/sqlfluff#3380]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

This valid (at least for redshift) query raises an L027 error when linting because sqlfluff can't resolve the mixed aliases.

```sql
SELECT
    account.id AS account_id,
    account.isdeleted AS is_deleted
FROM salesforce_sd.account
INNER JOIN salesforce_sd."user" ON salesforce_sd."user".id = account.ownerid

LIMIT 100
```

```text
❯ sqlfluff lint
== [/Users/alexkoch/repos/sqlfluff-test/test.sql] FAIL
L:   5 | P:  36 | L027 | Qualified reference 'salesforce_sd."user".id' not found
                       | in available tables/view aliases ['account', '"user"']
                       | in select with more than one referenced table/view.
All Finished 📜 🎉!
```

### Expected Behaviour

Query passes linting or is shown as a style type error.

### Observed Behaviour

Query fails linting with the "Qualified reference..." message when using different ways of aliasing how tables/columns are referenced with variations of `<<database>>.<<schema>>.<<table>>.<<column>>`

I've also seen a few other variations of L027 errors, particularly with DBT refs/source, but this seems the most general example I can come up with.

### How to reproduce

1. Install sqlfluff, create sqfluff config with:
```
[sqlfluff]
dialect = redshift
```
2. Create test.sql file:
```
SELECT
    account.id AS account_id,
    account.isdeleted AS is_deleted
FROM salesforce_sd.account
INNER JOIN salesforce_sd."user" ON salesforce_sd."user".id = account.ownerid

LIMIT 100
```
3. Run `sqlfluff lint`

### Dialect

redshift

### Version

❯ sqlfluff version
0.13.2
❯ python --version
Python 3.9.7

### Configuration

.sqlfluff:
```ini
[sqlfluff]
dialect = redshift
````

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
This issue was introduced by PR #2998. @dmohns: Would you be willing to look into this?

We may be able to fix this by using the functions `extract_possible_references()` and/or `extract_possible_multipart_references()`. (See rules L025 and L026 for rules similar to L027 which already use these functions.)

**Comment by dmohns** [comment]
I had a brief look at this, but `L025` and `L026` seem to rely on `SelectCrawler` a lot which I'm not very familiar with. 

For the short term: There are quite a few issues open right now related to checking table aliases as part of L027. Maybe it's worth considering to make checking table aliases exist a configurable option to L027?

**Comment by barrywhart** [comment]
@dmohns: Do you think you'll be able to fix it? If new code is not working, I'd rather fix or remove it than make it optional, unless we expect to fix it in the foreseeable future.

**Comment by barrywhart** [comment]
@dmohns: Unfortunately, we need to revert this change due to the multiple user issues reported.

If you like, I'd be happy to provide some guidance on creating a more robust implementation for a future release. Use of `SelectCrawler` or an equivalent approach that handles nested SQL statements will likely be necessary in order to accomplish this.
